docs/src/authorizations_processors/encoding.md

```diff
@@ -0,0 +1,56 @@
+# Encoding
+
+When we are passing messages between the `Authorization` contract and a `Processor` contract on a domain that is not using a `CosmWasm ExecutionEnvironment`, we need to encode the messages in a way that the `Processor` contract and the Libraries it calls can understand them. To do this we created two new contracts: `Encoder Broker` and `Encoder`.
+
+## Encoder Broker
+
+The `Encoder Broker` is a very simple contract that will route the messages to the correct `Encoder` contract. It is basically a mapping from `Encoder Version` to `Encoder Contract Address`. The `Encoder Broker` will be instantiated once on the `Main Domain` with an owner that can add/remove these mappings. An example of Mapping can be `"evm_encoder_v1"` to `<encoder_contract_address_on_neutron>`. The `Encoder Broker` has two queries: `Encode` and `Decode`, which routes the message to encode/decode to the `Encoder Version` specified.
+
+## Encoder
+
+The `Encoder` is the contract that will encode/decode the messages for a specific `ExecutionEnvironment`. It will be instantiated on the `Main Domain` an added to the `Encoder Broker` with a version/namespace. `Encoders` are defined for a specific `Execution Environment` and have an `Encode` and `Decode` query where we provide the Message to be encoded/decoded. Here's an example of how these queries look like:
```

Comment:
```suggestion
The `Encoder` is the contract that will encode/decode the messages for a specific `ExecutionEnvironment`. It will be instantiated on the `Main Domain` an added to the `Encoder Broker` with a version/namespace. `Encoders` are defined for a specific `Execution Environment` and have an `Encode` and `Decode` query where we provide the Message to be encoded/decoded. Here is an example of how these queries are performed:
```